paddle/phi/common/backend.h

```diff
@@ -130,6 +130,9 @@ inline std::ostream& operator<<(std::ostream& os, Backend backend) {
       os << "IPU";
       break;
     default: {
+      // As backend is of enum class type, backend can only takes
+      // the enumerated values, so what's the point of the default
+      // clause ???
       size_t device_type_id_ = static_cast<size_t>(backend) -
                                static_cast<size_t>(Backend::NUM_BACKENDS);
       std::string device_type = phi::GetGlobalDeviceType(device_type_id_);
```

Comment:
1. As backend is of enum class type, backend can only takes the enumerated values, so what's the point of the default
 clause ??? 
2. What's the point of `static_cast<size_t>(backend) -
                               static_cast<size_t>(Backend::NUM_BACKENDS);`